src/Admin/AbstractAdmin.php

```diff
@@ -1487,6 +1489,18 @@ public function attachAdminClass(FieldDescriptionInterface $fieldDescription)
 
             if (!$pool->hasAdminByClass($targetModel)) {
```

Comment:
Can't we use the `hasAdminByClass` method for this? IMHO the pool should now allow multiple admins for one entity

Reply:
How would you do this @core23 ?

Reply:
Im my project we have about 10 admin classes for one entity. 
Also im think that method getAdminByClass must be deprecated

Reply:
> Im my project we have about 10 admin classes for one entity.

That's not an issue. I have multiple classes for only one entity too.

> Also im think that method getAdminByClass must be deprecated

The `admin_code` would be required in that case then. Don't know the implication.
And there is another usage of the `adminByClass` method.

I don't see a bad thing to automatically get an admin by Class, if possible.

Reply:
Im discover this method using for getting admin in twig function: `sonata_urlsafeid` (it seams never used) and for attaching admins

Reply:
Im found `admin.generateUrl('edit', {'id' : object|sonata_urlsafeid(admin)` that can be replaced on `admin.generateObjectUrl('edit', object)`

Reply:
So. getAdminByClass can be be removed. Use getAdminByCode

Reply:
And `attachAdminClass` will throw own exception with suggest for define option `admin_code`.

Reply:
We could indeed deprecate `getUrlSafeIdentifier` in the TwigExtension. Do you mind creating the PR ?

But I'm against the deprecation of `getAdminByClass`, it's very useful when you only have one admin by class.

Reply:
We should kept method `getAdminClass` as legacy?

Reply:
For ex: (Instead of `getAdminByClass`)
![image](https://user-images.githubusercontent.com/30481080/96185795-1c7b4c00-0f43-11eb-90d1-a01dc1c742c1.png)